src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -200,21 +187,21 @@ internal ChangeSignatureResult ChangeSignature(Document document, int position,
                 declarationDocument, positionForTypeBinding, symbol, parameterConfiguration);
         }
 
-        private ChangeSignatureResult ChangeSignatureWithContext(ChangeSignatureAnalysisSucceededContext context, CancellationToken cancellationToken)
+        private Task<ChangeSignatureResult> ChangeSignatureWithContextAsync(ChangeSignatureAnalysisSucceededContext context, CancellationToken cancellationToken)
         {
             var options = GetChangeSignatureOptions(context);
             if (options == null)
             {
-                return new ChangeSignatureResult(succeeded: false);
+                return Task.FromResult(new ChangeSignatureResult(succeeded: false));
             }
 
-            return ChangeSignatureWithContext(context, options, cancellationToken);
+            return ChangeSignatureWithContextAsync(context, options, cancellationToken);
         }
 
-        internal ChangeSignatureResult ChangeSignatureWithContext(ChangeSignatureAnalysisSucceededContext context, ChangeSignatureOptionsResult options, CancellationToken cancellationToken)
+        internal async Task<ChangeSignatureResult> ChangeSignatureWithContextAsync(ChangeSignatureAnalysisSucceededContext context, ChangeSignatureOptionsResult options, CancellationToken cancellationToken)
         {
-            var succeeded = TryCreateUpdatedSolution(context, options, cancellationToken, out var updatedSolution);
-            return new ChangeSignatureResult(succeeded, updatedSolution, context.Symbol.ToDisplayString(), context.Symbol.GetGlyph(), options.PreviewChanges);
+            var (updatedSolution, confirmationMessage) = await CreateUpdatedSolutionAsync(context, options, cancellationToken).ConfigureAwait(false);
+            return new ChangeSignatureResult(updatedSolution != null, updatedSolution, context.Symbol.ToDisplayString(), context.Symbol.GetGlyph(), options.PreviewChanges, confirmationMessage: confirmationMessage);
```

Comment:
instead of `updatedSolution != null` here, can teh constructor of ChangeSignatureResult just do that itself?

Reply:
I could potentially, but there are many cases where we explicitly create a failure result and don't pass in any solution (we just pass succeeded: false).  I could modify it to just pass a null solution, but I'm not sure if that's much better

Reply:
ok.  i don't love this.  but it's not a big deal :)